client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -357,11 +357,11 @@ public void cancelled() {
     private void setHeaders(HttpRequest httpRequest, Header[] requestHeaders) {
         Objects.requireNonNull(requestHeaders, "request headers must not be null");
         for (Header defaultHeader : defaultHeaders) {
-            httpRequest.setHeader(defaultHeader);
+            httpRequest.addHeader(defaultHeader);
         }
         for (Header requestHeader : requestHeaders) {
             Objects.requireNonNull(requestHeader, "request header must not be null");
-            httpRequest.setHeader(requestHeader);
+            httpRequest.addHeader(requestHeader);
```

Comment:
I decided that it did make sense. This way you can, for instance, set the default user, then change it on a per-request basis.